Missing CMake Dependency

When building on a single core I get the following error:

```
cmake --build /home/erick/code/catalyst-latest/mlir//mlir-hlo/build --target check-mlir-hlo -j 1
[160/305] Building CXX object mhlo/analysis/CMakeFiles/obj.MhloTestAnalysis.dir/test_shape_component_analysis.cc.o
FAILED: mhlo/analysis/CMakeFiles/obj.MhloTestAnalysis.dir/test_shape_component_analysis.cc.o 
ccache /usr/bin/clang++ -DGTEST_HAS_RTTI=0 -D_DEBUG -D_GLIBCXX_ASSERTIONS -D_LIBCPP_ENABLE_ASSERTIONS -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/home/erick/code/catalyst-latest/mlir/llvm-project/llvm/include -I/home/erick/code/catalyst-latest/mlir/llvm-project/build/include -I/home/erick/code/catalyst-latest/mlir/llvm-project/mlir/include -I/home/erick/code/catalyst-latest/mlir/llvm-project/build/tools/mlir/include -I/home/erick/code/catalyst-latest/mlir/mlir-hlo -I/home/erick/code/catalyst-latest/mlir/mlir-hlo/build -I/home/erick/code/catalyst-latest/mlir/mlir-hlo/stablehlo -I/home/erick/code/catalyst-latest/mlir/mlir-hlo/build/stablehlo -fPIC -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -ffunction-sections -fdata-sections -O3 -DNDEBUG   -D_DEBUG -D_GLIBCXX_ASSERTIONS -D_LIBCPP_ENABLE_ASSERTIONS -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS  -fno-exceptions -funwind-tables -fno-rtti -UNDEBUG -std=gnu++17 -MD -MT mhlo/analysis/CMakeFiles/obj.MhloTestAnalysis.dir/test_shape_component_analysis.cc.o -MF mhlo/analysis/CMakeFiles/obj.MhloTestAnalysis.dir/test_shape_component_analysis.cc.o.d -o mhlo/analysis/CMakeFiles/obj.MhloTestAnalysis.dir/test_shape_component_analysis.cc.o -c /home/erick/code/catalyst-latest/mlir/mlir-hlo/mhlo/analysis/test_shape_component_analysis.cc
/home/erick/code/catalyst-latest/mlir/mlir-hlo/mhlo/analysis/test_shape_component_analysis.cc:24:10: fatal error: 'transforms/passes.h.inc' file not found
#include "transforms/passes.h.inc"
         ^~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.
make[1]: *** [Makefile:53: mhlo] Error 1
make[1]: Leaving directory '/home/erick/code/catalyst-latest/mlir'
make: *** [Makefile:44: mhlo] Error 2

```

Fixed with:

```diff
diff --git a/mhlo/analysis/CMakeLists.txt b/mhlo/analysis/CMakeLists.txt
index 4e15d73..a10d88d 100644
--- a/mhlo/analysis/CMakeLists.txt
+++ b/mhlo/analysis/CMakeLists.txt
@@ -13,6 +13,9 @@ add_mlir_library(MhloAnalysis
 add_mlir_library(MhloTestAnalysis
   test_shape_component_analysis.cc
 
+  DEPENDS
+  LMHLOTransformsPassIncGen
+
   LINK_COMPONENTS
   Core
 
```

--- a/tensorflow/compiler/xla/mlir_hlo/mhlo/analysis/CMakeLists.txt
+++ b/tensorflow/compiler/xla/mlir_hlo/mhlo/analysis/CMakeLists.txt
@@ -16,6 +16,9 @@ add_mlir_library(MhloTestAnalysis
   LINK_COMPONENTS
   Core
 
+  DEPENDS
+  LMHLOTransformsPassIncGen
+
   LINK_LIBS PUBLIC
   MLIRHLOAnalysis
   MLIRAnalysis

